SemiconductorTestLibrary.TestStandSteps/source/LeakageTest.cs

```diff
@@ -67,64 +70,80 @@ public static void LeakageTest(
-                            dcPowerPerPinOrPinGroup.ForceVoltage(voltageLevel, currentLimit);
+                            dcPowerPerPinOrPinGroup.ForceVoltage(voltageLevel, currentLimit, waitForSourceCompletion: true);
+                            dcPowerPerPinOrPinGroup.MeasureAndPublishVoltage("Leakage", out _);
+                            dcPowerPerPinOrPinGroup.PowerDown();
                         }
 
                         var digitalPins = tsmContext.FilterPinsByInstrumentType(new string[] { pinOrPinGroup }, InstrumentTypeIdConstants.NIDigitalPattern);
                         if (digitalPins.Any())
                         {
                             var digitalPerPinOrPinGroup = sessionManager.Digital(digitalPins);
-                            digitalPerPinOrPinGroup.ConfigureApertureTime(apertureTime);
                             digitalPerPinOrPinGroup.ForceVoltage(voltageLevel, currentLimit);
                             PreciseWait(settlingTime);
+                            digitalPerPinOrPinGroup.MeasureAndPublishVoltage("Leakage", out _);
+                            digitalPerPinOrPinGroup.TurnOffOutput();
                         }
                     }
                 }
                 else
                 {
+                    // Force specified voltage on all pins simultaneously.
                     InvokeInParallel(
-                        () =>
-                        {
-                            if (dcPower != null)
-                            {
-                                dcPower.ConfigureSourceDelay(settlingTime);
-                                dcPower.ConfigureMeasureSettings(dcPowerMeasureSettings);
-                                dcPower.ForceVoltage(voltageLevel, currentLimit, waitForSourceCompletion: true);
-                            }
-                        },
+                        () => dcPower?.ForceVoltage(voltageLevel, currentLimit, waitForSourceCompletion: true),
                         () =>
                         {
                             if (digital != null)
                             {
-                                digital.ConfigureApertureTime(apertureTime);
                                 digital.ForceVoltage(voltageLevel, currentLimit);
                                 PreciseWait(settlingTime);
                             }
                         });
-                }
 
-                // Measure current on all pins simultaneously.
-                InvokeInParallel(
-                    () =>
-                    {
-                        dcPower?.MeasureAndPublishCurrent("Leakage", out _);
-                    },
-                    () =>
-                    {
-                        digital?.MeasureAndPublishCurrent("Leakage", out _);
-                    });
+                    // Measure current on all pins simultaneously.
+                    InvokeInParallel(
```

Comment:
I guess there is no point of making MeasureAndPublishCurrent and PowerDown/TurnOffOutput in their own InvokeInParallel calls, now that they are separately handled in serial case? Moving them right after the ForceVoltage calls (like how you do it for serial case) should have better performance because there are less code merging points.

Reply:
@lornazh, the idea for the parallel case is to ensure that all pins are forced at the specified levels before performing the measurement. This is important for some DUTs, and ensures constancy for how pin groups are handled in the serial case.